lib/std/math/gdc.zig

```diff
@@ -0,0 +1,26 @@
+/// Find the greatest common divisor (GCD) of two integers.
+const std = @import("std");
+const expect = std.testing.expect;
+
+// Greatest common divisor
+fn gcd(a: anytype, b: anytype) @TypeOf(a) {
```

Comment:
This return type does technically work in this case, since `gcd(x, y) <= x`, but it's asymmetrical despite the logic itself being symmetrical (`gcd(x, y) == gcd(y, x)`). You should probably use `@TypeOf(a, b)` to perform peer type resolution (https://ziglang.org/documentation/master/#TypeOf).

Reply:
You could alternatively make the return type `Min(@TypeOf(a), @TypeOf(b))` since `gcd(x, y) <= x,y`, although note that you'll have to `@intCast` the return value if you do that (see next comment).

Reply:
`Min(@TypeOf(a), @TypeOf(b))` is not so good, because with the combination `gdc(0, 5000)` the return value is 5000. Therefore `@TypeOf(a, b)` is the best choice.